### PR TITLE
Scroll to load more

### PR DESCRIPTION
**Pull request by marcorei**
@puf as promised, my approach at adding scroll-to-load-more functionality to this lib. In my opinion it is still problematic that in order to add headers and/or footers you have to override onCreateViewHolder. Please let me know what you think.

### CODE REVIEW

**Comment by marcorei** [comment]
@googlebot done.

**Comment by puf** [comment]
Hey @marcorei , thanks for the awesome PR. Agreed that the API is still un-ideal, but this is definitely the cleanest infinite scroll I've seen on FirebaseUI. I have some comments, which I'll add in the code. 

I also don't think we're quite ready to add this functionality to FirebaseUI at this stage, so will keep this PR open for now.

File: library/src/main/java/com/firebase/ui/FirebaseArray.java [hunk 1]
```diff
@@ -64,6 +107,24 @@ public DataSnapshot getItem(int index) {
...
+        }
+        else if(mOrderASC == true){
+            mQuery = mOriginalQuery.limitToFirst(mCurrentSize);
+        }
+        else {
+            mQuery = mOriginalQuery.limitToLast(mCurrentSize);
+        }
+        setSyncing(true);
+        mQuery.addChildEventListener(this);
+        mQuery.addListenerForSingleValueEvent(this);
```
Comment (puf) [inline library/src/main/java/com/firebase/ui/FirebaseArray.java]: As you scroll, this keeps reloading the pervious items for each next page. This is a bandwidth concern, which unfortunately I think can only be solved by loading each "page" separately and then merging them into a single array "under the hood".

**Comment by marcorei** [comment]
Thank you for your feedback! Are you referring to the stage of the library or the stage of the PR?

File: library/src/main/java/com/firebase/ui/FirebaseArray.java (hunk 1 above)
Comment (marcorei) [inline library/src/main/java/com/firebase/ui/FirebaseArray.java]: I agree that the simplicity of this approach comes at the cost of bandwidth. Loading pages separately could result in a much more complex system (Managing listeners in order to receive updates for previously loaded pages or using startAt() in conjunction with moveable children in Queries could become issues). I'll have to think about how to solve this properly!

**Comment by puf** [comment]
Hey Marco. I meant the stage of the library, but the PR indeed would also need some iteration. Plus: we'd need to implement it in FirebaseUI-iOS too, which becomes a lot easier once we figure out the best API and implementation here first.

**Comment by samtstern** [comment]
@marcorei this is an old PR and the conversation seems to have died out so I am going to close it.  We appreciate your contribution here and it's very likely that a feature like this could be useful in the future.  If you're still interested, please open an issue and we can discuss.

**Comment by harshadura** [comment]
@puff @samtstern : wouldn't it be possible to release an external release of the new API after merging this code. IMO It would be very useful for amateur programmers who are using firebase and face this load-more issue.

PS: kudos @marcorei for the PR.

**Comment by puf** [comment]
We will only release infinite scroll/pagination support in FirebaseUI when
it's available on both Android and iOS.

But since this is an open-source project, you can fork the library and add
it. This old PR may serve as a great basis for that.

**Comment by harshadura** [comment]
Hello puff, like when you guys planning to support pagination for android.
Thanks

**Comment by puf** [comment]
No timelines. If you need it, I recommend forking the repo and rolling your
own.

**Comment by harshadura** [comment]
@ThunderRoid : Could you tell me how you have used this PR to enable scroll to load more feature. Thanks!

**Pull request closed without merging**